package.json

```diff
@@ -75,17 +76,31 @@
     "@types/jasminewd2": "^2.0.8",
     "@types/node": "~8.9.4",
     "codelyzer": "^5.1.2",
+    "husky": "^4.3.8",
     "jasmine-core": "~3.4.0",
     "jasmine-spec-reporter": "~4.2.1",
     "karma": "~4.1.0",
     "karma-chrome-launcher": "~2.2.0",
     "karma-coverage-istanbul-reporter": "~2.0.1",
     "karma-jasmine": "~2.0.1",
     "karma-jasmine-html-reporter": "^1.4.0",
+    "lint-staged": "^10.5.4",
+    "prettier": "^2.2.1",
     "protractor": "~7.0.0",
     "stylelint-config-standard": "^20.0.0",
     "ts-node": "~7.0.0",
     "tslint": "~5.15.0",
     "typescript": "~3.5.3"
+  },
+  "lint-staged": {
+    "*": [
+      "prettier --write",
+      "git add"
+    ]
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm run precommit && npx ng test --watch=false --progress=false --codeCoverage=true --browsers=ChromeHeadless"
```

Comment:
lets split this 
"pre-commit": npm run precommit,
 "pre-push": "ng test --no-watch --browsers=ChromeHeadless --codeCoverage=true"

Reply:
Done